pkg/listening-sockets/listening-sockets_test.go

```diff
@@ -0,0 +1,205 @@
+			Pod:       "",
+			Container: "test-container",
+			Optional:  false,
+		},
+		{
+			Direction: "Ingress",
+			Protocol:  "UDP",
+			Port:      500,
+			NodeRole:  "master",
+			Service:   "pluto",
+			Namespace: "",
+			Pod:       "",
+			Container: "test-container",
+			Optional:  false,
+		},
+	}
+)
+
+var _ = Describe("GenerateSS", func() {
+	// creating the fake clients, node, pods
+	BeforeEach(func() {
+		sch := runtime.NewScheme()
+
+		err := v1.AddToScheme(sch)
+		Expect(err).NotTo(HaveOccurred())
+
+		testNode := &v1.Node{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "test-node",
+				Labels: map[string]string{
+					"node-role.kubernetes.io/master": "",
+				},
+			},
+		}
+
+		fakeClient := fake.NewClientBuilder().WithScheme(sch).WithObjects(testNode).Build()
+		fakeClientset := fakek.NewSimpleClientset()
+
+		clientset = &client.ClientSet{
+			Client:          fakeClient,
+			CoreV1Interface: fakeClientset.CoreV1(),
+		}
+
+		ctrlTest = gomock.NewController(GinkgoT())
+		mockUtils = mock_utils.NewMockUtilsInterface(ctrlTest)
+	})
+
+	DescribeTable("should generate the correct ss tcp, udp output and the correct ssMatrix",
+		func(tc ssTestCase) {
+
+			// RunCommandOnPod had more that one calling and in each call we want other output
+			mockUtils.EXPECT().RunCommandOnPod(gomock.Any(), gomock.Any()).DoAndReturn(
+				func(pod *v1.Pod, cmd []string) ([]byte, error) {
+					switch {
+					case len(cmd) > 2 && strings.HasPrefix(cmd[2], "crictl ps -o json --id"):
+						return []byte(crictlExecCommandOut), nil // container data output
+
+					case strings.HasPrefix(cmd[2], "cat /proc/") &&
+						strings.Contains(cmd[2], "/cgroup"):
+						return []byte(procExecCommandOutput), nil // pid data output
+
+					case cmd[2] == "ss -anpltH":
+						return []byte(tcpExecCommandOutput), nil // tcp output
+
+					case cmd[2] == "ss -anpluH":
+						return []byte(udpExecCommandOutput), nil // udp output
+
+					default:
+						return nil, fmt.Errorf("unknown command")
+					}
+				}).AnyTimes()
+
+			mockUtils.EXPECT().
+				CreateNamespace(consts.DefaultDebugNamespace).
+				Return(nil).
+				AnyTimes()
+
+			mockUtils.EXPECT().
+				DeleteNamespace(consts.DefaultDebugNamespace).
+				Return(nil).
+				AnyTimes()
+
+			mockUtils.EXPECT().
+				CreatePodOnNode(gomock.Any(), consts.DefaultDebugNamespace, consts.DefaultDebugPodImage).
+				Return(mockPod, nil).AnyTimes()
+
+			mockUtils.EXPECT().DeletePod(mockPod).Return(nil).AnyTimes()
+
+			connectionCheck, err := NewCheck(clientset, mockUtils, "/some/dest/dir")
+			Expect(err).NotTo(HaveOccurred())
+
+			ssMat, ssOutTCP, ssOutUDP, err := connectionCheck.GenerateSS()
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(normalizeOutput(string(ssOutTCP))).To(Equal(normalizeOutput(tc.expectedTCPOutput)))
+			Expect(normalizeOutput(string(ssOutUDP))).To(Equal(normalizeOutput(tc.expectedUDPOutput)))
+			Expect(ssMat.Matrix).To(Equal(tc.expectedssMat))
+		},
+
+		Entry("Case 1", ssTestCase{
```

Comment:
I really like this format with ginkgo, it is much easier to read the test now. I think I was wrong with the describe table, don't think we have more tests to add here, so can just be an It